src/sdc11073/xml_types/xml_structure.py

```diff
@@ -819,27 +863,21 @@ def _compare_extension(left: etree_.ElementBase, right: etree_.ElementBase) -> b
     if len(left_children) == 0 and len(right_children) == 0:
         if left.text != right.text:  # mixed content is not allowed. only compare text if there are no children
             return False
-
     return all(map(_compare_extension, left_children, right_children))  # compare children but keep order
 
 
-class ExtensionLocalValue:
+class ExtensionLocalValue(list[etree_.ElementBase]):
 
     compare_method: Callable[[etree_.ElementBase, etree_.ElementBase], bool] = _compare_extension
     """may be overwritten by user if a custom comparison behaviour is required"""
 
-    def __init__(self, value: list[etree_.ElementBase] | None):
-        self.value = value or list()
-
-    def __eq__(self, other: ExtensionLocalValue) -> bool:
-        if not isinstance(other, self.__class__):
-            return False
-        if len(self.value) != len(other.value):
-            return False
-        for my_element, other_element in zip(self.value, other.value):
-            if not ExtensionLocalValue.compare_method(my_element, other_element):
+    def __eq__(self, other: Sequence) -> bool:
+        try:
+            if len(self) != len(other):
                 return False
-        return True
+        except AttributeError: # len of other cannot be determined
```

Comment:
when is this the case? If other is not a sequence then return `NotImplemented`

Reply:
check this [answer](https://github.com/Draegerwerk/sdc11073/pull/246#discussion_r1310146062)